### DISCUSSION TITLE
Are tests up to date?

### DISCUSSION THREAD

**Discussion by andodet** [hotosm/tasking-manager#4328] [Q&A] [answered]
I have just managed setting up my development environment using a slight variation of your `docker-compose` files and upon first running my tests I have noticed some error due to the wrong number of arguments passed to functions (example below)
```
======================================================================
ERROR: test_variable_replacing (unit.services.messaging.test_template_service.TestTemplateService)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/app/tests/backend/unit/services/messaging/test_template_service.py", line 15, in test_variable_replacing
    content = get_template("email_verification_en.html")
TypeError: get_template() missing 1 required positional argument: 'values'
```
Guess my question is: are tests up to date? Is it possible that methods got changed/updated without updating related tests?

Thanks for the clarification.

**Comment by willemarcel** [comment, marked as answer]
@andodet yes, some backend tests are outdated

**Reply by andodet** [reply]
Thanks @willemarcel, is there any issue/PR where tests updates are being discussed?

**Reply by willemarcel** [reply]
@andodet yesterday we merged #3407, which added a BaseTestCase. Recently we have been working to test some other parts https://github.com/hotosm/tasking-manager/pull/4071

If you're interested in work on this, your collaboration will be very helpful for us!

**Reply by andodet** [reply]
@willemarcel Sadly `docker-compose up` doesn't create the `test_tasking-environment` db on which the tests will get run on. Will try to figure where it does make sense to create that db in the Docker configs.

Also, I do believe that the entire Docker workflow should be refreshed as it still is a bit painful to get it to work properly.
